### ISSUE TITLE
CIFAR10 not having a license

### ISSUE THREAD

**Issue by psteinb** [carpentries-incubator/deep-learning-intro#445]
As just discussed [here](https://github.com/carpentries-lab/reviews/issues/25#issuecomment-1965113607) it might be worth considering to replace CIFAR-10 for some other dataset.

**Comment by psteinb** [comment]
I'd propose [fashionMNIST](https://github.com/zalandoresearch/fashion-mnist/blob/master/LICENSE).

**Comment by tobyhodges** [comment]
Thanks so much for the suggestion, @psteinb. As I mentioned on the review thread, I plan to spend some time trying to replace CIFAR-10 in the episode soon, hopefully next week. Your proposal seems like a great place to start. I will report back here on my progress...

**Comment by svenvanderburg** [comment]
+1 for fashionMNIST @tobyhodges let me know if you don't have time, than I will pick this up!

**Comment by svenvanderburg** [comment]
@tobyhodges I would suggest to first test things out in a (or multiple) jupyter notebook and have someone review it, before going into changing the actual lesson material.

**Comment by psteinb** [comment]
Re: fashionMNIST, while this is a nice dataset (easy to communicate about), there are two differences to CIFAR10:
- different shape 28x28 instead of 32x32
- fashionMNIST is greyscale instead of RGB
We should keep this in mind as it will affect teaching.

**Comment by tobyhodges** [comment]
I tried to work through the episode using fashion-MNIST as suggested by @psteinb. You can see my process and results in https://github.com/carpentries-incubator/deep-learning-intro/blob/testing_fashion_MNIST/fashion_MNIST.ipynb

A summary of my observations:

1. **Please check my working.** I might have implemented the convolutional layers incorrectly, in which case my results can be ignored! I needed to adjust the code to account for the single channel of the images, and I am not certain I did that correctly.
2. The differences between the results of the different kinds of NN do not appear to be as stark when applied to this dataset. As such I think there are a few statements, plus possibly the final challenge where we vary the dropout rate, that might no longer hold/be interesting (again, assuming my implementation was correct!)

**Comment by svenvanderburg** [comment]
@tobyhodges thank you for working on this! This looks great actually 👍 Seems like you are having fun with keras :)
1. That looks good
2. Yes, you are right that the differences are not very strong. It seems like the first simple model is actually a really good model that is hard to improve. The val accuracy of 0.856 of the first model is actually not beaten in the rest of the episode. I will see if I can tweak things a little bit to make this fit better in the storyline. Or we can try other datasets, for example the orginal mnist (but this is an even simpler problem, so I don't think it will help) or https://data.caltech.edu/records/mzrjq-6wc02 (seems to have a CC-BY 4.0 license but I suspect that the images are crawled).

**Comment by svenvanderburg** [comment]
OK, seems like the fashionMNIST dataset is a bit of a weird ML problem. It is very hard to overfit, and regularization only results in worse performance in the end;) I tried some approaches to force it into overfitting (bigger models, smaller dataset) but no good results.

I am looking into other datasets, but oh my god there are so many CC-BY licensed datasets that are actually just crawled from the web.

**Comment by colinsauze** [comment]
The dollar street dataset is 101GB! Is there a subset of it available?

**Comment by svenvanderburg** [comment]
@colinsauze I haven't found a subset, but I just downloaded it and if it works well with the lesson I will make a subset available with low-res pictures.

**Comment by tobyhodges** [comment]
If the license permits, we can publish a subset on FigShare, Zenodo, or similar.

**Comment by psteinb** [comment]
Use the dollar street sign dataset. Looking back at this conversation, 
this choice would not require too much adaptation with respect to text.

Best,
P

**Comment by tobyhodges** [comment]
This dollar street dataset sounds great to me, @svenvanderburg. Thanks so much for taking the time to explore it. 

One note: I saw you used PIL for loading and re-sizing the images. Would it be possible to switch over to using scikit-image (and imageio for the loading part)? That way we can point people to [DC Image Processing](https://datacarpentry.org/image-processing/) if they want to learn more about handling image data in Python?

**Comment by tobyhodges** [comment]
Would you like to hold a coworking session/sprint to prepare the updated episode? Or prefer to draft something yourself then ask others to review?

**Comment by svenvanderburg** [comment]
> This dollar street dataset sounds great to me, @svenvanderburg. Thanks so much for taking the time to explore it.
> 
> One note: I saw you used PIL for loading and re-sizing the images. Would it be possible to switch over to using scikit-image (and imageio for the loading part)? That way we can point people to [DC Image Processing](https://datacarpentry.org/image-processing/) if they want to learn more about handling image data in Python?

The data preprocessing will not be done in the course, the starting point of the episode will be to load the data in its preprocessed form. To focus on deep learning instead of image data wrangling.

**Comment by svenvanderburg** [comment]
> Would you like to hold a coworking session/sprint to prepare the updated episode? Or prefer to draft something yourself then ask others to review?

I want to get this through, it has been hanging for so long now. So I will draft something today. But I will organise a new sprint soon to pick up the remaining maintenance issues. I hope that's OK?

**Comment by svenvanderburg** [comment]
I'm working on it here: https://github.com/carpentries-incubator/deep-learning-intro/pull/448
The data is located here: https://zenodo.org/records/10837090

I hope to continue this next week, if anyone wants to pick this up in the meantime you are welcome! (Or start on transfer learning episode for example which would be really nice to add now).

**Comment by tobyhodges** [comment]
If you are happy for me to commit to your branch, @svenvanderburg, I can try to step in and make some further changes?
